tox_encryptsave/src/lib.rs

```diff
@@ -112,32 +104,30 @@ impl PassKey {
     use tox_crypto::pwhash::gen_salt;
     use tox_encryptsave::*;
 
-    assert_eq!(PassKey::with_salt(&[], gen_salt()), Err(KeyDerivationError::Null));
+    assert_eq!(PassKey::with_salt(&[], [42; SALT_LENGTH]).err(), Some(KeyDerivationError::Null));
     ```
     */
-    pub fn with_salt(passphrase: &[u8], salt: Salt) -> Result<PassKey, KeyDerivationError> {
+    pub fn with_salt(passphrase: &[u8], salt: [u8; SALT_LENGTH]) -> Result<PassKey, KeyDerivationError> {
         if passphrase.is_empty() { return Err(KeyDerivationError::Null) };
 
         let passhash = Sha256::digest(passphrase);
-        let OpsLimit(ops) = OPSLIMIT_INTERACTIVE;
-        let mut key = secretbox::Key([0; secretbox::KEYBYTES]);
-
-        let maybe_key = PrecomputedKey::from_slice({
-            // securely zeroes the key when `kb` goes out of scope
-            let secretbox::Key(ref mut kb) = key;
-            derive_key(
-                kb,
-                &passhash,
-                &salt,
-                OpsLimit(ops * 2),
-                MEMLIMIT_INTERACTIVE
-            ).or(Err(KeyDerivationError::Failed))?
-        });
-
-        let salt = Box::new(salt);
-        let key = Box::new(maybe_key.ok_or(KeyDerivationError::Failed)?);
-
-        Ok(PassKey { salt, key })
+        let mut key = [0; xsalsa20poly1305::KEY_SIZE];
```

Comment:
Wait a minute! Can we decrypt old profile with the new code?

Reply:
Yes :)

Reply:
Do we get the same results as before?

Reply:
Yes, we have a test for this.